selftests/functional/test_basic.py

```diff
@@ -858,6 +858,51 @@ def tearDown(self):
         shutil.rmtree(self.tmpdir)
 
 
+class RunnerSimpleTestStatus(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+
+        self.config_file = script.TemporaryScript('avocado.conf',
+                                                  "[simpletests.status]\n"
+                                                  "warn_regex = ^WARN$\n"
+                                                  "skip_regex = ^SKIP$\n")
+        self.config_file.save()
+
+        self.warn_script = script.TemporaryScript('avocado_warn.sh',
+                                                  "#!/bin/sh\necho WARN",
+                                                  'avocado_simpletest_'
+                                                  'functional')
+        self.warn_script.save()
+
+        self.skip_script = script.TemporaryScript('avocado_warn.sh',
+                                                  "#!/bin/sh\necho SKIP",
+                                                  'avocado_simpletest_'
+                                                  'functional')
+        self.skip_script.save()
+
+        os.chdir(basedir)
+
+    def test_simpletest_warn(self):
+        cmd_line = ('%s --config %s run --job-results-dir %s --sysinfo=off'
+                    ' %s --json -' % (AVOCADO, self.config_file.path,
+                                      self.tmpdir, self.warn_script.path))
+        result = process.system_output(cmd_line, ignore_status=True)
+        json_results = json.loads(result)
+        self.assertEquals(json_results['tests'][0]['status'], 'WARN')
+
+    def test_simpletest_skip(self):
```

Comment:
Hmm, have you considered merging those two into a single test? For unittests it's useful to have better granularity but here it means 1s vs. 0.6s...